### PR TITLE
Nemwel School Library Milestone 2 : Person, Student, Teacher classes

### PR DESCRIPTION
**Pull request by Nemwel-Boniface**
## Hello reviewer(s) :wave: 
In this milestone, I was able to implement the knowledge learned on OOP in Ruby by completing the first setup for the application. 
I was able to accomplish the following:

- [x] Set up the Person, Student, and Teacher classes
- [x] Create both public and private methods in the `Person` superclass
- [x] Implement Encapsulation by implementing the constructor `initialize` which had several instance variables
- [x] Implement inheritance by extending the properties and methods from our super class `Person` to the child classes `Student` and `Teacher` classes
- [x] Implement polymorphism by overriding the inherited method `can_use_services?` in the `Teacher` class setting it to `true` 
- [x] Update the project's README.md file including instructions on how to run the project's linters
- [x] Check and fix all offenses that were found

Thank you for taking the time to review my project :pray: 
**In case I missed something or in the PR description confused something (OOP application) I would appreciate it if you point that one out to me**

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  # getters
+  attr_reader :id
+  attr_accessor :name, :age
```
Comment (Deepakdanger) [inline person.rb]: - [x] The getter & setters should be defined after the class is defined, It is a good way of writing code. Kindly shift these lines to line 2.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  # Setters
+
+  def can_use_services?
+    if @age >= 18 || @parent_permission
```
Comment (Deepakdanger) [inline person.rb]: - [x] here we have to use the function `of_age?` . Kindly use it. 
```
def can_use_services? 
    is_of_age?() || @parent_permission
  end

```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Nemwel-Boniface, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Nemwel-Boniface) [reply person.rb]: Thank you @Deepakdanger  for this suggestion. I have effected this required change

File: person.rb (hunk 2 above)
Comment (Nemwel-Boniface) [reply person.rb]: Thank you @Deepakdanger for this suggestion. I have effected the requested change.

**Review by Meltrust — APPROVED** [review]
Hi @Nemwel-Boniface,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for updating your method, and for correcting your class getters and setters✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**